### PR TITLE
Upgrade eslint from v8 to v9

### PR DESCRIPTION
**Pull request by digitliofficial**
https://github.com/cybersemics/em/issues/2320

1. updated eslint v8 to v9
2. updated eslint configuration structure on new pattern
3. all lint issues resolved
4. test is passed
5. work on both windows and mac os
6. some eslint dependencies also updated.
7. added some little new config as per client request like {args: none} unused var

### CODE REVIEW

File: tsconfig.eslint.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "extends": "./tsconfig.json",
+    "include": [
+      "**/*.ts",
+      "**/*.tsx",
+      "**/*.js",
+      "capacitor.config.ts",
+      "panda.config.ts",
+      "scripts/**/*.ts",
+      "vite.config.ts",
+      "vitest.workspace.ts"
```
Comment (raineorshine) [inline tsconfig.eslint.json:11]: I think you can remove the individual files since `**/*.ts` covers them all.

**Review by raineorshine — CHANGES_REQUESTED** [review]
Thank you! I think this is coming together well now. Just a few questions for you, and hopefully will get this merged soon.

File: src/shortcuts.ts [hunk 2]
```diff
@@ -282,43 +306,17 @@ export const inputHandlers = (store: Store<State, any>) => ({
     shortcutEmitter.trigger('shortcut', shortcut)
 
     if (!shortcut.canExecute || shortcut.canExecute(store.getState())) {
-      if (!shortcut.permitDefault) e.preventDefault()
+      e.preventDefault()
```
[the remark this answers is not in the document]
Comment (digitliofficial) [reply src/shortcuts.ts:309]: yes fix it in new commit.

File: src/actions/__tests__/importText.ts [hunk 3]
```diff
@@ -746,7 +746,6 @@ p.p1 {margin: 0.0px 0.0px 0.0px 0.0px; font: 12.0px 'Helvetica Neue'}
   })
 
   it('should paste text properly that is copied from WebStorm', () => {
-    /* eslint-disable no-irregular-whitespace */
```
[the remark this answers is not in the document]
Comment (digitliofficial) [reply src/actions/__tests__/importText.ts:749]: this command define in many place in this file. but after enable rule in eslint config. it throws error on only 3 places. rest all are unused. so thats why we remove from there.

File: src/util/__tests__/validateRoam.ts [hunk 4]
```diff
@@ -70,10 +70,10 @@ test('it returns false if there are any missing properties', () => {
   type requiredKeyTypes = 'create-email' | 'uid' | 'create-time' | 'string'
   const requiredKeys = ['create-email', 'uid', 'create-time', 'string']
   requiredKeys.forEach(key => {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const invalidRoamString = JSON.stringify([
       {
         ...testData[0],
+        //disable lint error because using destructuring to get the rest.
```
[the remark this answers is not in the document]
Comment (digitliofficial) [reply src/util/__tests__/validateRoam.ts:76]: it was showing error previoustly.  but now it fixed. and also remove this line in new commit.

File: tsconfig.eslint.json (hunk 1 above)
Comment (digitliofficial) [reply tsconfig.eslint.json]: yes, removed.

**Review by raineorshine — APPROVED** [review]
We got there! Thanks for your persistence 🙏

**Review by raineorshine — CHANGES_REQUESTED** [review]
Oops, I spoke a moment too soon. There's a warning on yarn install I just noticed. Can you investigate? The expectation is that eslint and all eslint-related plugins are upgraded and all peer dependency issues are resolved.

```
➤ YN0060: │ eslint is listed by your project with version 9.14.0 (pd90f6), which doesn't satisfy what @typescript-eslint/eslint-plugin and other dependencies request (^8.57.0).
```

**Comment by digitliofficial** [comment]
> Oops, I spoke a moment too soon. There's a warning on yarn install I just noticed. Can you investigate? The expectation is that eslint and all eslint-related plugins are upgraded and all peer dependency issues are resolved.
> 
> `➤ YN0060: │ eslint is listed by your project with version 9.14.0 (pd90f6), which doesn't satisfy what @typescript-eslint/eslint-plugin and other dependencies request (^8.57.0).`

I have looked into this so it turns out that if I resolve this warning then it gives more dependent warnings one after the other. All of these are warnings not errors.

Another important thing here is when I update the above plugin some other plugin gives a warning but at some point I am stuck with this plugin eslint-config-react-app plugin. The reason is its an old plugin that is last updated 3 years ago. This is all very common when you go after such warnings.

**Review by raineorshine — APPROVED** [review]

**Pull request merged**